elfi/methods/post_processing.py

```diff
@@ -0,0 +1,270 @@
+        will be adjusted.
+
+        Returns
+        -------
+          a Sample object containing the adjusted posterior
+        """
+        outputs = {}
+        for (i, name) in enumerate(self.parameter_names):
+            theta_i = self.sample.outputs[name][self._finite[i]]
+            adjusted = self._adjust(i, theta_i, self.regression_models[i])
+            outputs[name] = adjusted
+
+        res = results.Sample(method_name=self._name, outputs=outputs,
+                             parameter_names=self._parameter_names)
+        return res
+
+    def _adjust(self, i, theta_i, regression_model):
+        """Adjust a single parameter using a fitted regression model.
+
+        Parameters
+        ----------
+        i : int
+          the index of the parameter
+        theta_i : np.ndarray
+          a vector of parameter values to adjust
+        regression_model
+          a fitted regression model
+
+        Returns
+        -------
+        adjusted_theta_i : np.ndarray
+          an adjusted version of the parameter values
+        """
+        raise NotImplementedError
+
+    def _input_variables(self, model, sample, summary_names):
+        """Construct a matrix of regressors.
+
+        Parameters
+        ----------
+        model : elfi.ElfiModel
+          the inference model
+        sample
+          a sample object from an ABC algorithm
+        summary_names : list[str]
+          names of the summary nodes
+
+        Returns
+        -------
+        X
+          a numpy array of regressors
+        """
+        raise NotImplementedError
+
+    def _get_finite(self):
+        # TODO: Access the variables through the getters
+        finite_inputs = np.isfinite(self._X).all(axis=1)
+        finite = [finite_inputs & np.isfinite(self._sample.outputs[p])
+                  for p in self._parameter_names]
+        all_finite = all(map(all, finite))
+        self._finite = finite
+        if not (all(finite_inputs) and all_finite):
+            warnings.warn("Non-finite inputs and outputs will be omitted.")
+
+
+class LinearAdjustment(RegressionAdjustment):
+    """Regression adjustment using a local linear model."""
+    _regression_model = LinearRegression
+    _name = 'LinearAdjustment'
+
+    def __init__(self, **kwargs):
+        super(LinearAdjustment, self).__init__(**kwargs)
+
+    def _adjust(self, i, theta_i, regression_model):
+        b = regression_model.coef_
+        return theta_i - self.X[self._finite[i], :].dot(b)
+        
+    def _input_variables(self, model, sample, summary_names):
+        """Regress on the differences to the observed summaries."""
+        observed_summaries = np.stack([model[s].observed
+                                       for s in summary_names], axis=1)
+        summaries = np.stack([sample.outputs[name]
+                              for name in summary_names], axis=1)
+        return summaries - observed_summaries
+
+
+def adjust_posterior(model, sample, summary_names,
```

Comment:
At least please change the order so that sample is the first one. In the future I'd like to remove the dependency to model.